### ISSUE TITLE
nx release in fixed mode not incrementing the release version correctly of applications

### ISSUE THREAD

**Issue by pgraemer** [nrwl/nx#21872]
### Current Behavior

Given an integrated repo with two libraries (`lib-a` and `lib-b`) and one app (`app-a`)
`lib-a` is a dependency of `app-a` . Only `app-a` is included in the release projects using conventional commits.

After a change (fix) in `lib-a`, when running nx release version, nx does not increment the release version for `app-a`.

### Expected Behavior

Nx should increment the release version for `app-a` using `lib-a` with changes.

### GitHub Repo

https://github.com/pgraemer/nx-examples

### Steps to Reproduce

1. clone repo
2. introduce a change using conventional commit in `libs/cart/cart-page`
3. run `nx release --skip-publish --dry-run` 
4. check version increment

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node   : 20.11.0
   OS     : darwin-arm64
   yarn   : 1.22.19
   
   nx (global)        : 18.0.4
   nx                 : 18.0.4
   @nx/js             : 18.0.4
   @nx/jest           : 18.0.4
   @nx/linter         : 18.0.4
   @nx/eslint         : 18.0.4
   @nx/workspace      : 18.0.4
   @nx/angular        : 18.0.4
   @nx/cypress        : 18.0.4
   @nx/devkit         : 18.0.4
   @nx/eslint-plugin  : 18.0.4
   @nx/react          : 18.0.4
   @nrwl/tao          : 18.0.4
   @nx/web            : 18.0.4
   @nx/webpack        : 18.0.4
   typescript         : 5.3.3
   ---------------------------------------
   Community plugins:
   @ngrx/component-store : 17.0.1
   @ngrx/effects         : 17.0.1
   @ngrx/entity          : 17.0.1
   @ngrx/router-store    : 17.0.1
   @ngrx/store           : 17.0.1
   @ngrx/store-devtools  : 17.0.1
```

### Package Manager Version

yarn 1.22.19

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

as `nx release` is still very early i am unsure if this is expected behaviour. removing the limit to only release e.g. `app-a` results in a correct version increment of all libs but not apps.

**Comment by knoefel** [comment]
Any news on this? We're facing the same issue right now when making a change to a lib.

**Comment by CantGetRight82** [comment]
Hi, i would also like to know your thoughts on this. I have started a monorepo in nx, that currently only has 2 apps and 2 libs.

After updating one library:
- `nx affected` shows 2 apps and 1 lib
- `nx release version` shows 0 apps and 0 libs

I think `nx release` should use the same logic as `nx affected` when it comes to deciding what has been updated.

I could maybe help out with a pull request, but would like to know what the direction is on this matter.

Bonus: it might be possible to use a dependency graph to detect if the changes are solely related to specific imported units, thereby being able to skip apps that only rely on other parts of the library?
